first-aid.lic

```diff
@@ -8,24 +8,46 @@ class FirstAid
   include DRC
 
   def initialize
+    settings = get_settings
     @chart_data = get_data('anatomy-charts').first_aid_charts
-
-    train_first_aid
+	@textbook = settings.textbook
+    if @textbook
+      @booktype = 'textbook'
+	  textbook_charts
+    else
+      @booktype = 'compendium'
+	  compendium_charts
+	end
   end
-
-  def train_first_aid
+  
+  def compendium_charts
     bput('get my compendium', 'You get')
     bput('look my compendium', '^The compendium lies open to the section on .* physiology')
     pause
     charts = reget(40).grep(/^   .+$/).map(&:strip)
 
     charts_to_read = charts.select { |chart| @chart_data[chart]['scholarship'] <= DRSkill.getrank('Scholarship') }
-
+	study_charts(charts_to_read)
+  end
+  
+  def textbook_charts
+    charts_to_read = []
+    bput('get my textbook', 'You get')
+    readable_charts = @chart_data.select { |chart| @chart_data[chart]['scholarship'] <= DRSkill.getrank('Scholarship')}
```

Comment:
This `selects` (and line 44) should be changed to use `|name, info|` like line 37. There's no reason to index into `@chart_data` when the iterator `chart` has the same information.